Week3/homework/step2-6.js

```diff
@@ -1,16 +1,16 @@
 'use strict';
 
+const flatten = array => [].concat(...array);
+
 const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  return flatten(arr);
 }
-
+// flatten any arr
 function flattenArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  return flatten(arr.map(el => (Array.isArray(el) ? flattenArray3d(el) : el)));
```

Comment:
Really clever use of recursion here - and it's great that you're checking Array first using the ternary operator. You're really demonstrating a good understanding of the language!

Reply:
Thank you.